pkg/process/util/orchestrator/pod.go

```diff
@@ -43,26 +44,31 @@ func ProcessPodlist(podList []*v1.Pod, groupID int32, cfg *config.AgentConfig, h
 		// extract pod info
 		podModel := extractPodMessage(podList[p])
 
+		// static pods "uid" are actually not unique across nodes.
+		// we differ from the k8 uuid format in purpose to differentiate those static pods.
+		if pod.IsStaticPod(podList[p]) {
+			newUID := generateUniqueStaticPodHash(hostName, podList[p].Name, podList[p].Namespace, clusterName)
+			// modify it in the original pod for the YAML and in our model
+			podList[p].UID = types.UID(newUID)
+			podModel.Metadata.Uid = newUID
+		}
+
+		pd := podList[p]
+		if skip := apiserver.SkipKubernetesResource(pd.UID, pd.ResourceVersion); skip {
```

Comment:
Moved this, because we can reuse the UID check here

Reply:
Thanks for the catch 👍 . Somehow I overlooked because this was initially used for debugging 💯